AX: Merge AccessibilityAttachment into AccessibilityRenderObject

hoffmanjoshua opened the pull request:
#### 8beef5fc0982dc02e4984c97c525e1947afeea0e
<pre>
AX: Merge AccessibilityAttachment into AccessibilityRenderObject
<a href="https://bugs.webkit.org/show_bug.cgi?id=297400">https://bugs.webkit.org/show_bug.cgi?id=297400</a>
<a href="https://rdar.apple.com/158325218">rdar://158325218</a>

Reviewed by Tyler Wilcock.

Merge the logic from AccessibilityAttachment into AccessibilityRenderObject
and AccessibilityNodeObject as part of our effort to flatten our class
hierarchy.

A new helper class, AXAttachmentHelpers, was added to encapsulate the
HTMLAttachmentElement-specific logic.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/accessibility/AXAttachmentHelpers.cpp: Renamed from Source/WebCore/accessibility/AccessibilityAttachment.cpp.
(WebCore::AXAttachmentHelpers::hasProgress):
(WebCore::AXAttachmentHelpers::accessibilityText):
* Source/WebCore/accessibility/AXAttachmentHelpers.h: Renamed from Source/WebCore/accessibility/AccessibilityAttachment.h.
* Source/WebCore/accessibility/AXCoreObject.h:
(WebCore::AXCoreObject::hasProgress const):
* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::AXObjectCache::createObjectFromRenderer):
* Source/WebCore/accessibility/AXSearchManager.cpp:
* Source/WebCore/accessibility/AccessibilityNodeObject.cpp:
(WebCore::AccessibilityNodeObject::valueForRange const):
(WebCore::AccessibilityNodeObject::hasProgress const):
(WebCore::AccessibilityNodeObject::accessibilityText const):
* Source/WebCore/accessibility/AccessibilityNodeObject.h:
* Source/WebCore/accessibility/AccessibilityObject.h:
(WebCore::AccessibilityObject::isAttachmentElement const):
* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
(WebCore::AccessibilityRenderObject::isAttachmentElement const):
(WebCore::AccessibilityRenderObject::computeIsIgnored const):
(WebCore::AccessibilityRenderObject::determineAccessibilityRole):
* Source/WebCore/accessibility/AccessibilityRenderObject.h:
* Source/WebCore/accessibility/ios/WebAccessibilityObjectWrapperIOS.mm:
(-[WebAccessibilityObjectWrapper accessibilityValue]):
(-[WebAccessibilityObjectWrapper accessibilityIsAttachmentElement]):

Canonical link: <a href="https://commits.webkit.org/298740@main">https://commits.webkit.org/298740@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/37c361162fcc579a4d5fbc118599b16cf08ab7f5

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/116395 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/36056 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/26607 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/122452 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/66956 "Built successfully") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/118284 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/36750 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/44644 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/88401 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/42875 "Build was cancelled. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/119344 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/29304 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/104425 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/68838 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/28385 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/22528 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/66133 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/98686 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/22688 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/125601 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/43289 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/32507 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/97109 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/43654 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/100630 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/96904 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24682 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/42193 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/20099 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/39243 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/43177 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/48768 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/42643 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/45983 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/44348 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

twilco commented on Source/WebCore/accessibility/AXAttachmentHelpers.cpp:
```diff
@@ -25,66 +25,30 @@
-    auto& progressString = getAttribute(progressAttr);
+    auto& progressString = attachmentElement.getAttribute(progressAttr);
     bool validProgress;
     float result = std::max<float>(std::min<float>(progressString.toFloat(&validProgress), 1), 0);
```
I know this code existed before, but if progressString.toFloat executes a code path that fails to set `validProgress`, we'll return an uninitialized bool from this function, which is undefined behavior. Can we set `validProgress` to a default value of `false` to be safe?

hoffmanjoshua replied:
Yes, good find! I'll update this.

hoffmanjoshua committed:
AX: Merge AccessibilityAttachment into AccessibilityRenderObject
https://bugs.webkit.org/show_bug.cgi?id=297400
rdar://158325218

Reviewed by Tyler Wilcock.

Merge the logic from AccessibilityAttachment into AccessibilityRenderObject
and AccessibilityNodeObject as part of our effort to flatten our class
hierarchy.

A new helper class, AXAttachmentHelpers, was added to encapsulate the
HTMLAttachmentElement-specific logic.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/accessibility/AXAttachmentHelpers.cpp: Renamed from Source/WebCore/accessibility/AccessibilityAttachment.cpp.
(WebCore::AXAttachmentHelpers::hasProgress):
(WebCore::AXAttachmentHelpers::accessibilityText):
* Source/WebCore/accessibility/AXAttachmentHelpers.h: Renamed from Source/WebCore/accessibility/AccessibilityAttachment.h.
* Source/WebCore/accessibility/AXCoreObject.h:
(WebCore::AXCoreObject::hasProgress const):
* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::AXObjectCache::createObjectFromRenderer):
* Source/WebCore/accessibility/AXSearchManager.cpp:
* Source/WebCore/accessibility/AccessibilityNodeObject.cpp:
(WebCore::AccessibilityNodeObject::valueForRange const):
(WebCore::AccessibilityNodeObject::hasProgress const):
(WebCore::AccessibilityNodeObject::accessibilityText const):
* Source/WebCore/accessibility/AccessibilityNodeObject.h:
* Source/WebCore/accessibility/AccessibilityObject.h:
(WebCore::AccessibilityObject::isAttachmentElement const):
* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
(WebCore::AccessibilityRenderObject::isAttachmentElement const):
(WebCore::AccessibilityRenderObject::computeIsIgnored const):
(WebCore::AccessibilityRenderObject::determineAccessibilityRole):
* Source/WebCore/accessibility/AccessibilityRenderObject.h:
* Source/WebCore/accessibility/ios/WebAccessibilityObjectWrapperIOS.mm:
(-[WebAccessibilityObjectWrapper accessibilityValue]):
(-[WebAccessibilityObjectWrapper accessibilityIsAttachmentElement]):

Canonical link: https://commits.webkit.org/298740@main
